components/query_filter/utils_unittest.cc

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/query_filter/utils.h"
+
+#include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+#include "url/gurl.h"
+
+TEST(BraveQueryFilter, FilterQueryTrackers) {
+  EXPECT_EQ(MaybeApplyQueryStringFilter(
+                GURL("https://brave.com"), GURL("https://brave.com"),
+                GURL("https://test.com/?gclid=123"), "GET", false),
+            GURL("https://test.com/"));
+  EXPECT_EQ(MaybeApplyQueryStringFilter(
+                GURL("https://brave.com"), GURL("https://brave.com"),
+                GURL("https://test.com/?fbclid=123"), "GET", false),
+            GURL("https://test.com/"));
+  EXPECT_EQ(MaybeApplyQueryStringFilter(
+                GURL("https://brave.com"), GURL("https://brave.com"),
+                GURL("https://test.com/?mkt_tok=123"), "GET", false),
+            GURL("https://test.com/"));
+  EXPECT_EQ(
+      MaybeApplyQueryStringFilter(
+          GURL("https://brave.com"), GURL("https://brave.com"),
+          GURL("https://test.com/?gclid=123&unsubscribe=123"), "GET", false),
+      GURL("https://test.com/?unsubscribe=123"));
+  EXPECT_EQ(
+      MaybeApplyQueryStringFilter(
+          GURL("https://brave.com"), GURL("https://brave.com"),
+          GURL("https://test.com/?gclid=123&Unsubscribe=123"), "GET", false),
```

Comment:
I think you meant to use the `mkt_tok` parameter here because this test case is the same as the previous one.

Reply:
My goal was to make the tests pass as they were passing before (as a check I didn't break anything) But with the knowledge I gained during this process I'm sure I can add some valid test comments here to help everyone understand what is going on.

I should also add some more tests which I will do. 
Though I don't know how to run these tests locally yet. so I didn't touch anything.

Reply:
I added some more tests with a variety of situations